lib/dynamoid/fields.rb

```diff
@@ -368,7 +369,7 @@ def set_created_at
     # @since 0.2.0
     def set_updated_at
       # @_touch_record=false means explicit disabling
-      if self.class.timestamps_enabled? && !updated_at_changed? && @_touch_record != false
+      if self.class.timestamps_enabled? && !changed? && @_touch_record != false
```

Comment:
!updated_at_changed? -> !changed?
because ActiveRecord modifies updated_at field only when changed? is true.

Reply:
The idea here is that we don't override `updated_at` provided by user:

```ruby
model.update_attributes(updated_at: Time.now - 1.month)
```

Reply:
Thanks for your quick feedback. Absolutely.

now I changed the condition like this;
`if self.class.timestamps_enabled? && changed? && !updated_at_changed? && @_touch_record != false`

When...
```
・changed? is false  
=> updated_at will NOT be updated

・updated_at provided by user 
=> updated_at will NOT be updated(override) because changed? is true but !updated_at_changed? is false

・changed? is true and updated_at not provided by user 
 => updated_at will be updated
```